cmake/Modules/FindQuadMath.cmake

```diff
@@ -46,4 +46,37 @@ if(QuadMath_FOUND AND NOT TARGET QuadMath::QuadMath)
   target_compile_options(QuadMath::QuadMath INTERFACE
     $<$<CXX_COMPILER_ID:GNU>:-fext-numeric-literals>
   )
+
+  # Check for numeric_limits specialization
+  cmake_push_check_state()
+  set(CMAKE_REQUIRED_LIBRARIES quadmath)
+  set(CMAKE_REQUIRED_FLAGS)
+  if(${CMAKE_CXX_COMPILER_ID} STREQUAL GNU)
+    set(CMAKE_REQUIRED_FLAGS "-fext-numeric-literals")
+  endif()
+  set(CMAKE_REQUIRED_FLAGS "${CMAKE_REQUIRED_FLAGS} -D_GLIBCXX_USE_FLOAT128")
+  check_cxx_source_compiles("
+  #include <quadmath.h>
+  #include <limits>
+  #include <type_traits>
+
+  template<bool> struct dependent_false : std::false_type {};
+  template<bool T>
+  void check()
+  {
+      if constexpr (std::numeric_limits<__float128>::is_specialized) {
+          ;
+      } else {
+          static_assert(dependent_false<T>::value);
+      }
+  }
+
+  int main ()
+  {
+      check<false>();
+  }" QuadMath_HAS_LIMITS)
```

Comment:
I'm probably missing something, but if you "only" check whether or not something compiles, might you not get away with the simpler source code
```C++
#include <limits>
void f()
{
    static_assert(std::numeric_limits<__float128>::is_specialized);
}
```
instead?  The compiler explorer [godbolt.org](https://godbolt.org/#z:OYLghAFBqd5TKALEBjA9gEwKYFFMCWALugE4A0BIEAZgQDbYB2AhgLbYgDkAjF%2BTXRMiAZVQtGIHgBYBQogFUAztgAKAD24AGfgCsp5eiyahUAUgBMAIUtXyKxqiIEh1ZpgDC6egFc2TEAB2QPJ3ABkCJmwAOT8AI2xSEABmcgAHdCViFyYvX38gkIys5yEIqNi2BKTUh2wnHJEiFlIiPL8A4PtsR1KmJpaicpj4xJT7Ztb2gq6lSaHIkaqx5IBKe3QfUlROLktkyNRfHABqM2SPegI2YiVz3DMtAEFHp4A3dAJME5oIVbPAlYTnMWM5UAB9FhKFStCBzTAgEBMPyJAgQq43Ih3C7g8E0ejoUE8CwADnuiIISnBSjS9QIEgIAC9sJhVucgWZAgARLjrejcACs/ACXB05HQ3A8tlswM222wZwsyT45CI2l56wA1ikAGwAOmkgQsFgAnDwtALpCSLDxkgLDNxpPw2CABVpyCKxRKuPwlCB3WrRbzyHBYCgMGw0gxEpRqBGo4wksAlKw0kokOgiAIGEREn6IHF1eQ4pEWgBPbgqkusUhlgDycV09UDKojHGEdaY9ArQfIODiPmAHgk9D9vH4ODYxmAkl7hFIzYIb2wY7F2HU9R8ucr/Eiuf5vaucVI5a8OCLRFI1x35GXpDimWwXOwU5MVxM6vW%2BJYyYAagRsAAdzrWkRRVQRhDECROBkORhGUNRNF7fQLEMac0GlaxDAIOI/UgdZ0DSPoxwAWjrZIThIgB1EdKKo9dLxYSjaUwYwwUo%2BFzi5cxrFsHhAn4dA7yvHA8L%2BbpehyNwmE8bwOgMcJFkqaoDGKbIhGmAIeHSTJ1KYYZlLGbS6gaIQBimOSCmMnpFzM%2BYDNGJJjPmTSDBBVoHOWJz1iUOUdikPlBWFItvROdQSR1EidWkE5gFQVATggS8fCYTV/ggfBiDIRVlVWfhAx0VZ1iQbAWBwJJxIPJ1yBdN0PRC7hfX9VVPy1FIeD1AUdRJE0tB1HgdWSZJAmkAULHtA9kmC3tvXyz8Q0QUNkDQdBI2jCgqAgeN1rQIwTHBZLUuzehc1IfNC17atyxvK7awbJsnBvNtmCITtuyLftB2HehRxvSdp1nMV50XZdV34ddN23cdKGEHoiyPE9azPXYVUva9obvB8VGfV8Z0iUAgy/Iw/wA4DQJvCDRHESRYMphCNCLfRtL20xMLsI8xIIoiclI8i6No6iGJPZiWTYtEOKITAuJ4mxrAsLQJcwI8uKPGXbEE4SvhXeAfJs0yAggdxXO0xSKkc1TdL6Y2dJKHJPJU6zJLswZrZMvpzIWM2vLclzLK0iZBntoyfL8mDAq4IV6pm7gwoiqKYqOadEsOtLEsykhSByng8pawm2uSEk9Qi6QtECAUBv4k0zQdLhqtq91PUExr7GagrgyWiBw1WhMY027bExAFo2DeEkSMTkwSJT47TvOotbp7KtSzuxtmye1b21erseyB7AByHEcxxVf6312IGCAXBpQaLCHUC3FHd1hg8xQR08sHv1UrxdDHEixp8XwB/G81vwkyAiBZgFN5BQRprIOmKgGbIVdGhD8assIcx1uKbmQheYUWogLei6hGIi1YsIcWJFOLJG4mzeWitlYUI5pLFBVgNaJBEtrfCElbIGyNn7BSMkg5ORtnpa2ak%2Bj8Lcnrd2vt8j%2Bzdo0eySlzbORdjwpRHkFHe2zhsLY/lNEHkjo3cUMch4jxIkxcewBk6kBSqnDKhAM5Zxzm3IqNc66ugbg1H0LcAytXINqZIWhOoWGkHafiFgdThP6oaGuU0o5embk4ha8AO5dzWomWMW1u47WADIbSdATp5moBdMU88bpL3rCvR60Nnodi3h9XeX0D5/X/ifD658Qba17DfO%2BRY9xw0PDhRGZZkYXk/jeTGj4cYAI/ITAQxMlD/lAeTaGlMoEwRgfIemSExT6FSCzDCvFUE4U5hg4i3AyI4Joj9OiQsmIkRYmLBKZDJbSyoV6TWol0GyNcIbGS1tTZLAdoIq2KigV23UYCr5/QpHyUdpwj2YjVFtBBe5T2ALg5aPlAFGu%2BiPGx0itFWK8UTgyD1DwNOdjsr7GznNPO5ASplTGOJPxFgDTEn4jqLQ40tAyC0CaFxzo3GxKbp4v03iZnJJWqk3ucZMkDzeKgNIaRwRvB4CacERhcxzHBOoaK08CkFjnmU0pNZykPRbPwapm93pznqfvH6h8JzNJnKfCcbTL4dLXBuW%2BUMVS9KfvwF%2BSM34jPRiqcZ2MnXvgJoVWZP55mkzAWBfgKzqZrLgooOBWy9AgFkHsxh2FcLoMIqcrgAB6OsFgTilvhPs2WVh5bMNIKw45kLpKyWkbwzACLQUaRBSIsFXsIUSLkcojtsL9bwvBUZAOFkx0ztRYZbyGKdHhxxdHLgYUCVsCUPKk4KqTR6g1SuIg5KsqZypY4nx9LyrUHDq4uqBjZpeNzoVfOgQ9S9R1CNG0vKTQChNENaJ004meKceHCwwHhU0tfbePMUlpBAA%3D) seems to suggest that that would be sufficient.

Reply:
Yeah, sometimes u get stuck on an initial train of thought.. can definitely be simplified.

Reply:
done.